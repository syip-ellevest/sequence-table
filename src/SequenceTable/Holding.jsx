/**
 * Renders information for a holding in a transaction object or sequencable
 * that contains one, e.g. a Buy or Sell order.
 * 
 */
import * as React from 'react'

const Holding = (props) => {
  const {
    symbol,
    shareQuantity,
  } = props.holding
  
  return (
    <div>
      <p>{symbol}</p>
      <p>Share</p>
      <p>{shareQuantity}</p>
    </div>
  )
}

// Proptypes: symbol (str), share (str)
// default props: isMobile false 

export default Holding