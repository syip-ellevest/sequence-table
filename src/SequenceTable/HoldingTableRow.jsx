import * as React from 'react'

const HoldingTableRow = (props) => {
  const {
    symbol,
    share,
  } 
  return (
    <tr>

    </tr>
  )
}

export default HoldingTableRow