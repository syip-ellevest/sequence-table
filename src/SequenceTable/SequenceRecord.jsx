/**
 * Renders a sequence record. 
 * @param {obj} record 
 * 
 */
import * as React from 'react'

const SequenceRecord = (props) => {
  const {
    date, 
    type, 
    status,
  } = props.record

  return (
    <div>
      <p>{date}</p>
      <p>{type}</p>
      <p>{status}</p>
      {props.children}
    </div>
  )
}

export default SequenceRecord