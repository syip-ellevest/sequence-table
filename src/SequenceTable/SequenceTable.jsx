/**
 * Represents a transaction sequence in a tabular format. 
 * Each record in the sequence is represented as a table body
 * (<tbody>), which is a group of table rows (<trs>).
 * 
 * To better understand the backend system that this view is representing, see 
 * docs on Transaction Sequences:
 * https://ellevest.atlassian.net/wiki/spaces/ENG/pages/62394411/Transaction+Sequences
 * https://github.com/Ellevest/transaction_system
 * 
 * @param {array} sequence - array of record objects; a record is a "step" in
 * the sequence. 
 * 
 */

import * as React from 'react'
import SequenceMobileTableBody from './mobile/SequenceMobileTableBody'
import HoldingMobileTableRow from './mobile/HoldingMobileTableRow'

const SequenceTable = (props) => {
  const renderTableBodies = (record) => {
    return (
      <SequenceMobileTableBody record={record}>
        { // Render holdings if available as nested rows
          record.holdings
          && record.holdings.map((holding) => {
            return <HoldingMobileTableRow holding={holding} />
          })   
        }
      </SequenceMobileTableBody>
    )
  }

  const tableBodies = props.sequence.map(
    record => renderTableBodies(record)
  )

  return (
    <table>
      {tableBodies}
    </table>
  )
}

export default SequenceTable