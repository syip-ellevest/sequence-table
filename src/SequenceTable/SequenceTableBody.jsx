/**
 * Takes one transaction object & renders into a <tbody> 
 * @desktop
 * 
 * @param {obj} record 
 */
import * as React from 'react'

const SequenceTableBody = (props) => {
  const {
    date, type, status,
  } = props.record

  return (
    <tbody>
      <tr>
        <td>
          <p>{date}</p>
        </td>
        <td>
          <p>{type}</p>
        </td>
        <td>
          <p>{status}</p>
        </td>
      </tr>
      {props.children}
    </tbody>
  )
}

export default SequenceTableBody