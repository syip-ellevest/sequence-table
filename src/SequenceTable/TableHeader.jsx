/**
 * Header for an table component.
 */

import * as React from 'react'

/**
 * 
 * @param {array} columns - array of column objects to be represented in this
 * table.
 * Sample column object:
    {
      key: string representing the key in the table data,
      title: display title for this column
    }
 */
const renderHeaderCells = (columns) => {
  return columns.map(column => <th>{column.title}</th> )
}

const TableHeader = (props) => {
  return (
    <thead>
      <tr>
        {renderHeaderCells(props.columns)}
      </tr>
    </thead>
  )
}

export default TableHeader