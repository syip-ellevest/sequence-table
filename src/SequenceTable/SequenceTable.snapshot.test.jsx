import React from 'react'
import SequenceTable from './SequenceTable'
import renderer from 'react-test-renderer'

import acatsTestSequence from '../testData/acats-sequence'

describe('Sequence Table', function () {
	it('Matches its mobile snapshot', function () {
		const tree = renderer.create(<SequenceTable sequence={acatsTestSequence} />)
		expect(tree).toMatchSnapshot()
	})
})