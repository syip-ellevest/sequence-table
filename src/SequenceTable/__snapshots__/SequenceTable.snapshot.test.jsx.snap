// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sequence Table Matches its mobile snapshot 1`] = `
<table>
  <tbody>
    <tr>
      <td
        colspan="2"
      >
        <p>
          07/25/2017
        </p>
      </td>
    </tr>
    <tr>
      <td>
        <p>
          Rebalance
        </p>
      </td>
      <td>
        <p>
          Pending
        </p>
      </td>
    </tr>
  </tbody>
  <tbody>
    <tr>
      <td
        colspan="2"
      >
        <p>
          07/22/2017
        </p>
      </td>
    </tr>
    <tr>
      <td>
        <p>
          ACATS
        </p>
      </td>
      <td>
        <p>
          Success
        </p>
      </td>
    </tr>
    <tr>
      <td>
        <p>
          SSTK
        </p>
        <p>
          Share
        </p>
        <p>
          1000
        </p>
      </td>
    </tr>
    <tr>
      <td>
        <p>
          VTI
        </p>
        <p>
          Share
        </p>
        <p>
          10000
        </p>
      </td>
    </tr>
    <tr>
      <td>
        <p>
          TSLA
        </p>
        <p>
          Share
        </p>
        <p>
          10
        </p>
      </td>
    </tr>
  </tbody>
  <tbody>
    <tr>
      <td
        colspan="2"
      >
        <p>
          07/16/2017
        </p>
      </td>
    </tr>
    <tr>
      <td>
        <p>
          Transfer Process Initiated
        </p>
      </td>
      <td>
        <p />
      </td>
    </tr>
  </tbody>
  <tbody>
    <tr>
      <td
        colspan="2"
      >
        <p>
          07/11/2017
        </p>
      </td>
    </tr>
    <tr>
      <td>
        <p>
          Stock rest and settle
        </p>
      </td>
      <td>
        <p>
          Complete
        </p>
      </td>
    </tr>
  </tbody>
</table>
`;
